ui/src/components/Main.jsx

```diff
@@ -54,6 +55,7 @@ function Main() {
       )}
 
       <SelectProposalContainer />
+      <ClearQueueDialog />
```

Comment:
Sure, this dialog only makes sense in the context of the _Samples_ page, but it doesn't really matter since its state is managed globally. I prefer to have all the dialogs in one place. It might uncover opportunities to simplify the global state and combine a few dialogs together into more general abstractions.

Reply:
Yes, it would be nice to only have, if possible, one main dialog component